sdk/python/feast/pyspark/launchers/standalone/local.py

```diff
@@ -22,6 +22,10 @@
     StreamIngestionJobParameters,
 )
 
+# In-memory cache of Spark jobs
+# This is necessary since we can't query Spark jobs in local mode
+JOB_CACHE = {}
```

Comment:
I'd check that python grpc server doesn't use a process pool or fork or something like that, so this is indeed the only copy of JOB_CACHE. I know in e.g. uWSGI it may be an issue, since it runs multiple python processes for you.

Reply:
I haven't found any related documentation regarding multiprocessing & grpc. Also, when testing things locally I didn't see inconsistent outputs in list_jobs. I should have seen different answers on different requests if it was using multiprocessing.